fragment.go

```diff
@@ -972,22 +1078,44 @@ func (f *fragment) rangeNEQ(bitDepth uint, predicate uint64) (*Row, error) {
 	return b, nil
 }
 
-func (f *fragment) rangeLT(bitDepth uint, predicate uint64, allowEquality bool) (*Row, error) {
-	keep := NewRow()
-
+func (f *fragment) rangeLT(bitDepth uint, predicate int64, allowEquality bool) (*Row, error) {
 	// Start with set of columns with values set.
-	b := f.row(uint64(bitDepth))
+	b := f.row(bsiExistsBit)
+
+	// Create predicate without sign bit.
+	upredicate := uint64(predicate)
+	if predicate < 0 {
+		upredicate = uint64(-predicate)
+	}
+
+	// If predicate is positive, return all positives less than predicate and all negatives.
+	if (predicate >= 0 && allowEquality) || (predicate >= -1 && !allowEquality) {
+		pos, err := f.rangeLTUnsigned(b.Difference(f.row(bsiSignBit)), bitDepth, upredicate, allowEquality)
+		if err != nil {
+			return nil, err
+		}
+		neg := b.Intersect(f.row(bsiSignBit))
```

Comment:
Do we need this intersection? In the set value code, I believe we're clearing the sign bit for any row that we're clearing the existence bit for. If so, the sign bit is already effectively intersected with the existence bit.